src/Microsoft.TestPlatform.CrossPlatEngine/TestRunner/ProxyStartTestRunnerManager.cs

```diff
@@ -41,15 +43,21 @@ public void StartTestRunner(StartTestRunnerCriteria criteria, IStartTestRunnerEv
         {
             var session = new Session();
 
-            // Should be done by spawning a task.
+            var taskList = new List<Task>();
             while (parallelLevel-- > 0)
             {
-                var operationManagerProxy = this.proxyCreator();
-                operationManagerProxy.SetupChannel(criteria.Sources, criteria.RunSettings);
+                taskList.Add(Task.Factory.StartNew(
+                    () =>
+                    {
+                        var operationManagerProxy = this.proxyCreator();
+                        operationManagerProxy.SetupChannel(criteria.Sources, criteria.RunSettings);
 
-                TestRunnerPool.Instance.AddProxy(session, operationManagerProxy);
+                        TestRunnerPool.Instance.AddProxy(session, operationManagerProxy);
+                    }));
             }
 
+            Task.WaitAll(taskList.ToArray());
```

Comment:
You can create an array of the length equals to  `parallelLevel` and send it to `Task.WaitAll` instead of having list and converting it to array.

Reply:
Done.